numpy/_core/_add_newdocs.py

```diff
@@ -2676,10 +2676,13 @@
 
 add_newdoc('numpy._core.multiarray', 'ndarray', ('nbytes',
     """
-    Total bytes consumed by the elements of the array.
+    Total number of bytes the array's data would consume if stored
+    contiguously in memory.
 
     Notes
     -----
+    If the array is a view, this shows how much memory it *would* use
+    if it were copied into a separate array.
     Does not include memory consumed by non-element attributes of the
     array object.
```

Comment:
Maybe we ca add that it also doesn't include memory indirectly held by the elements.
(I.e. if you store Python objects or the new `StringDType`)